File: src/Atlasd/Battlenet/Exceptions/ClientException.cs
using System;
using System.Net;

namespace Atlasd.Battlenet.Exceptions
{
    class ClientException : Exception
    {
        public ClientState Client { get; private set; }

        public ClientException(ClientState client) : base()
        {
            Client = client;
        }

        public ClientException(ClientState client, string message) : base(message)
        {
            Client = client;
        }

        public ClientException(ClientState client, string message, Exception innerException) : base(message, innerException)
        {
            Client = client;
        }
    }
}

File: src/AtlasV/Battlenet/Exceptions/ClientException.vb
Imports System
Imports System.Net

Namespace AtlasV.Battlenet.Exceptions
    Class ClientException
        Inherits Exception

        Public Property Client As ClientState

        Public Sub New(ByVal varClient As ClientState)
            MyBase.New()
            Client = varClient
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String)
            MyBase.New(varMessage)
            Client = varClient
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String, ByVal varInnerException As Exception)
            MyBase.New(varMessage, varInnerException)
            Client = varClient
        End Sub
    End Class
End Namespace
